hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleSystemContractOperations.java

```diff
@@ -92,45 +95,37 @@ public HandleSystemContractOperations(@NonNull final HandleContext context, @Nul
 
     @Override
     public ContractCallStreamBuilder externalizePreemptedDispatch(
-            @NonNull final TransactionBody syntheticBody, @NonNull final ResponseCodeEnum preemptingStatus) {
+            @NonNull final TransactionBody syntheticBody,
+            @NonNull final ResponseCodeEnum preemptingStatus,
+            @NonNull final HederaFunctionality functionality) {
         requireNonNull(syntheticBody);
         requireNonNull(preemptingStatus);
+        requireNonNull(functionality);
 
         return context.savepointStack()
-                .addChildRecordBuilder(ContractCallStreamBuilder.class)
+                .addChildRecordBuilder(ContractCallStreamBuilder.class, functionality)
                 .transaction(transactionWith(syntheticBody))
                 .status(preemptingStatus);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void externalizeResult(
-            @NonNull final ContractFunctionResult result, @NonNull final ResponseCodeEnum responseStatus) {
-        final var childRecordBuilder = context.savepointStack().addChildRecordBuilder(ContractCallStreamBuilder.class);
-        childRecordBuilder
-                .transaction(Transaction.DEFAULT)
-                .contractID(result.contractID())
-                .status(responseStatus)
-                .contractCallResult(result);
-    }
-
     @Override
     public void externalizeResult(
             @NonNull final ContractFunctionResult result,
             @NonNull final ResponseCodeEnum responseStatus,
-            @NonNull Transaction transaction) {
+            @NonNull final Transaction transaction) {
         requireNonNull(transaction);
         context.savepointStack()
-                .addChildRecordBuilder(ContractCallStreamBuilder.class)
+                .addChildRecordBuilder(ContractCallStreamBuilder.class, CONTRACT_CALL)
                 .transaction(transaction)
                 .status(responseStatus)
                 .contractCallResult(result);
     }
 
     @Override
-    public Transaction syntheticTransactionForNativeCall(Bytes input, ContractID contractID, boolean isViewCall) {
+    public Transaction syntheticTransactionForNativeCall(
+            @NonNull final Bytes input, @NonNull final ContractID contractID, boolean isViewCall) {
```

Comment:
I guess as long as you are adding `final`s one more for isViewCall?  Obviously very much a nit.